### PR TITLE
py-blosc2, -ndindex: submission

### PR DESCRIPTION
**Pull request by someuser12**
#### Description

Submitting `py-blosc2` and its dependency `py-ndindex`.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.7 19H2026 x86_64
macOS 11.7.4 20G1120 x86_64
macOS 12.6.3 21G419 arm64

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: python/py-ndindex/Portfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+checksums           rmd160  3e3fc8e48bc295567e49b3a37190c89a9009dfca \
+                    sha256  bf9bd0b76eeada1c8275e04091f8291869ed2b373b7af48e56faf7579fd2efd2 \
+                    size    85371
+
+if {${name} ne ${subport}} {
+    depends_build-append \
+                        path:bin/cython:py${python.version}-cython
+    depends_test-append \
+                        port:py${python.version}-hypothesis \
+                        port:py${python.version}-numpy \
```
Comment (reneeotten) [inline python/py-ndindex/Portfile:31]: NumPy is listed by upstream under "extras_require" with the comment "# NumPy is only required when using array indices"; do you think this is a commonly used thing by end-users? If so, I would probably recommend to add it to `depends_lib-append` so that it will be available at runtime.

File: python/py-blosc2/Portfile [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-blosc2
+version             2.2.0
+revision            0
+categories-append   devel
+license             BSD
+
+python.versions     37 38 39 310 311
```
Comment (reneeotten) [inline python/py-blosc2/Portfile:12]: it [looks](https://github.com/Blosc/python-blosc2/blob/043eab4d63d97efa6b43d95b6876b2b5914a96fe/setup.py#L55) like that Python 3.7 is not supported?

File: python/py-blosc2/Portfile [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    depends_build-append \
+                        port:ninja \
+                        port:py${python.version}-numpy \
+                        port:py${python.version}-scikit-build
+    depends_lib-append  path:bin/cmake:cmake \
+                        port:py${python.version}-cpuinfo \
+                        path:bin/cython:py${python.version}-cython \
+                        port:py${python.version}-msgpack \
+                        port:py${python.version}-ndindex \
+                        port:py${python.version}-rich
```
Comment (reneeotten) [inline python/py-blosc2/Portfile:53]: the dependencies don't look completely correct. From what I gather how upstream lists thing is that:
1. numpy is also a runtime dependency
2. cython and cmake should only be a build dependency
3. setuptools is also listed as build dependency

File: python/py-blosc2/Portfile [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    post-patch {
+        reinplace "s|, \"cmake\", \"ninja\"||" ${worksrcpath}/pyproject.toml
```
Comment (reneeotten) [inline python/py-blosc2/Portfile:56]: this seems an unnecessary change as you do actually need both of them... or?

File: python/py-blosc2/Portfile [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+homepage            https://www.blosc.org/python-blosc2/python-blosc2.html
+
+checksums           rmd160  e23ddcafb6a0e2d6da0fb4377e81d7c6227178d8 \
+                    sha256  4cf3893fe85da4bd899ecf7a457fd09b2212fad256182100108de7a92b88aa87 \
+                    size    4135248
+
+if {${name} ne ${subport}} {
+    # Exception: py-cpuinfo currently only works on X86 and some ARM CPUs.
+    #supported_archs     i386 x86_64
```
Comment (reneeotten) [inline python/py-blosc2/Portfile:42]: if this is commented out because it isn't true or redundant please remove

**Review by reneeotten — CHANGES_REQUESTED** [review]

**Comment by reneeotten** [comment]
also, the `py-blosc` is being build with optimizations (e.g., AVX2 / SSE4 and such). That should not be the case as it might lead to inconsistencies in the buildbot generated binary and the end users' system. Typically, one would add a `native` variant to enable such optimizations which would anyway require the port to be build on the end users' machine.

File: python/py-ndindex/Portfile (hunk 1 above)
Comment (someuser12) [reply python/py-ndindex/Portfile]: Thanks @reneeotten for your review. I followed your recommendation.

File: python/py-blosc2/Portfile (hunk 2 above)
Comment (someuser12) [reply python/py-blosc2/Portfile]: Indeed, officially Python 3.7 is not supported, so I removed it (anyway the end of life of Python 3.7 is in June).

File: python/py-blosc2/Portfile (hunk 3 above)
Comment (someuser12) [reply python/py-blosc2/Portfile:53]: Thanks for checking this. Dependencies should now be correct.

File: python/py-blosc2/Portfile (hunk 5 above)
Comment (someuser12) [reply python/py-blosc2/Portfile]: I removed this (which was taken over from `py-blosc`'s Portfile). `py-blosc2` was successfully tested on an M1 CPU, but I don't have access to more exotic CPUs.

**Comment by someuser12** [comment]
Regarding the build with optimizations, this is work in progress. I think some `CMakeLists.txt` files in subdirectories need to be patched to disable them.

**Comment by reneeotten** [comment]
ping @someuser12

**Comment by pmetzger** [comment]
@someuser12 Can we finish this off?

**Comment by pmetzger** [comment]
@someuser12 It would be a shame to close this without finishing it?

**Comment by pmetzger** [comment]
@someuser12 I'm going to close this for now. Please feel free to re-open it if you wish to work on it again.
